smdebug/tensorflow/keras.py

```diff
@@ -1059,12 +1074,15 @@ def _on_any_batch_end(self, batch, mode, logs=None):
                     start_step=self.mode_steps[mode],
                 )
 
+    @error_handler.catch_smdebug_errors()
     def on_train_batch_end(self, batch, logs=None):
         self._on_any_batch_end(batch, ModeKeys.TRAIN, logs=logs)
 
+    @error_handler.catch_smdebug_errors()
     def on_test_batch_end(self, batch, logs=None):
         self._on_any_batch_end(batch, ModeKeys.EVAL, logs=logs)
 
+    @error_handler.catch_smdebug_errors()
     def on_predict_batch_end(self, batch, logs=None):
         self._on_any_batch_end(batch, ModeKeys.PREDICT, logs=logs)
 
```

Comment:
Why isn't a decorator necessary for `wrap_optimizer()`?

Reply:
The scope of this change is to cover all smdebug functions called during training with the default smdebug configuration (from the AWS TF codebase essentially).

`wrap_optimizer` would only be called by the user manually and would thus be considered a custom smdebug configuration, so we don't wrap `wrap_optimizer` with the error handler.

Reply:
Clarified this in the PR description too.

Reply:
I see, thanks for clarifying.